### PR TITLE
4.1 Add hardware debug pin and print

### PR DESCRIPTION
**Pull request by andyp1per**

### CODE REVIEW

**Comment by Hwurzburg** [comment]
@andyp1per I merged the runcam stuff...thanks....then updated this PR to only be for the debug stuff in the DEV section....still needs a few questions answered....thanks

**Comment by Hwurzburg** [comment]
also, Andy has said this is an alternative or maybe its a requirement?:
optional support for hard-fault debugging using soft-serial output to a pin
  To use this setup a pin like this:

    Pxx FAULT OUTPUT HIGH

  for some pin Pxx

**Review by tridge — APPROVED** [review]

File: dev/source/docs/debugging.rst [hunk 1]
```diff
@@ -66,9 +66,18 @@ The stderr and stdout file handles are not available on ChibiOS, so fprintf is n
...
 .........................
 
+
 This output appears on the console.  For ChibiOS-based boards this is the first-defined SERIAL device.
 
 
+GPIO Assert Debugging
+..............
+
+It is possible to output debug assertions from ChibiOS (so the debug errors that you get if you build with --enable-asserts) using a GPIO pin. To do this you need to set "HAL_GPIO_PIN_FAULT" in hwdef.dat to the GPIO pin you want to use and then connect a logic analyzer to the pin. You will also need to configure your logic analyzer to interpret the serial stream and print readable information but most modern logic analyzers are able to do this automatically.
```
Comment (tridge) [inline dev/source/docs/debugging.rst:76]: you should just set FAULT pin in hwdef.dat, the HAL_GPIO_PIN_ is automatically added by the generator

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by Hwurzburg** [comment]
fixed build error,squashed....merging

**Pull request merged**